src/stores/app/version.js

```diff
@@ -329,6 +328,88 @@ export default class AppVersionStore extends Store {
   }
 
   @action
+  async fetchActiveVersions(params = {}) {
+    const defaultParams = {
+      limit: this.maxLimit
+    };
+
+    this.isLoading = true;
+    const result = await this.request.get(
+      'active_app_versions',
+      assign(defaultParams, params)
+    );
+    this.versions = get(result, 'app_version_set', []);
+    this.totalCount = get(result, 'total_count', 0);
+    this.isLoading = false;
+  }
+
+  @action
+  fetchTypeVersions = async appId => {
+    this.isLoading = true;
+    const result = await this.request.get('app_versions', {
+      limit: this.maxLimit,
+      app_id: appId
+    });
+    const versions = get(result, 'app_version_set', []);
+
+    // get all unique version types
+    const types = _.uniq(versions.map(item => item.type));
+    // get type relatived versions
+    const typeVersions = [];
+    types.forEach(type => typeVersions.push({
+      type,
+      versions: versions.filter(item => item.type === type)
+    }));
+    this.typeVersions = typeVersions;
+
+    this.isLoading = false;
+  };
+
+  checkPackageFile = file => {
+    const result = true;
+    const maxsize = 2 * 1024 * 1024;
+    this.createError = '';
+
+    if (!/\.(tar|tar\.gz|tar\.bz|tgz)$/.test(file.name.toLocaleLowerCase())) {
+      this.createError = ts('file_format_note');
+      return false;
+    }
+    if (file.size > maxsize) {
+      this.createError = ts('The file size cannot exceed 2M');
+      return false;
+    }
+
+    return result;
+  };
+
+  uploadPackage = async (base64Str, file) => {
+    console.log(file);
+    this.isLoading = true;
+    this.packageName = file.name;
+
+    const result = await this.request.post('apps/validate_package', {
+      version_type: this.version.type,
+      version_package: base64Str
+    });
+    this.isLoading = false;
+
+    if (result.error_details) {
+      this.uploadStatus = 'error';
+      this.createError = ts('Upload_Package_Error');
+      this.uploadError = result.error_details;
+      return;
+    }
+    if (result.error || result.err) {
```

Comment:
No need to check error / err, ask backend devs, i think check err is enough

Reply:
ok